EmbeddedStock/Controllers/ComponentController.cs

```diff
@@ -49,5 +49,27 @@ public IActionResult Show(int id)
             var model = new ComponentShowViewModel(component, componentType);
             return View(model);
         }
+
+        public IActionResult Delete(int id)
+        {
+            ViewData["id"] = id;
+            return View();
+        }
+
+        [ActionName("Delete")]
+        [HttpPost]
+        public IActionResult Delete_Component(int id)
+        {
+            var component = FindComponentById(id);
+            context.Components.Remove(component);
+            context.SaveChanges();
+
+            return RedirectToAction("Show", "ComponentType", new { id = component.ComponentTypeId });
+        }
+
+        private Component FindComponentById(int id)
+        {
+            return context.Components.Single(ct => ct.ComponentId == id);
```

Comment:
måske lidt forvirrende at variablen hedder "ct", det er bare sukker, men det var jo oprindeligt for "componentType"